File: app/src/androidTest/java/app/we/go/emojidraw/features/practice/StubPracticePresenter.java
package app.we.go.emojidraw.features.practice;

import java.util.List;

import javax.inject.Inject;

import app.we.go.emojidraw.arch.mvp.BasePresenter;
import app.we.go.emojidraw.model.Stroke;

/**
 * Stub implementation of {@link PracticeContract.Presenter} that:
 * 1) notifies the view that all emojis guessed correctly on the 1st stroke
 * 2) if user has cheated (skipped), notifies the view accordingly
 * 2) calls the expired callback when it receives the signal that time expired.
 */
public class StubPracticePresenter extends BasePresenter<PracticeContract.View>
        implements PracticeContract.Presenter {


    private boolean cheated;

    @Inject
    public StubPracticePresenter() {

    }


    @Override
    public void onStrokeAdded(List<Stroke> strokes) {
        if (getView() != null && !strokes.isEmpty()) {
            if (cheated) {
                getView().onAllEmojisDrawnWithCheat();
            } else {
                getView().onAllEmojisDrawn();
            }
        }
    }

    @Override
    public void onTimeExpired() {
        if (getView() != null) {
            getView().onTimeOut();
        }
    }

    @Override
    public void onSkip() {
        cheated = true;
    }
}

File: app/src/androidTest/java/app/we/go/emojidraw/features/practice/StubPracticePresenter.kt
package app.we.go.emojidraw.features.practice

import app.we.go.emojidraw.arch.mvp.BasePresenter
import app.we.go.emojidraw.model.Stroke
import javax.inject.Inject

/**
 * Stub implementation of [PracticeContract.Presenter] that:
 * 1) notifies the view that all emojis guessed correctly on the 1st stroke
 * 2) if user has cheated (skipped), notifies the view accordingly
 * 2) calls the expired callback when it receives the signal that time expired.
 */
class StubPracticePresenter @Inject
constructor() : BasePresenter<PracticeContract.View>(), PracticeContract.Presenter {

    private var cheated: Boolean = false

    override fun onStrokeAdded(strokes: List<Stroke>) {
            if (strokes.isNotEmpty()) {
                if (cheated) {
                    view?.onAllEmojisDrawnWithCheat()
                } else {
                    view?.onAllEmojisDrawn()
                }
            }
    }

    override fun onTimeExpired() {
        view?.onTimeOut()
    }

    override fun onSkip() {
        cheated = true
    }
}
